lib/require-from-worker.js

```diff
@@ -1,43 +1,19 @@
 'use strict';
 
-const debugGenerator = require('heimdalljs-logger');
-const hashForDep = require('hash-for-dep');
 const AstPlugins = require('./ast-plugins');
-
-const _logger = debugGenerator('ember-cli-htmlbars-inline-precompile');
+let astPlugin = {};
 
 module.exports = {
   build(options) {
-    let pluginInfo = this.setupDependentPlugins(options.parallelConfig);
-
-    return AstPlugins.setup(pluginInfo, {
-      templateCompilerPath: options.templateCompilerPath,
-      useCache: true,
-    });
-  },
-
-  setupDependentPlugins(parallelConfig) {
-    const plugins = [];
-    const cacheKeys = [];
-
-    parallelConfig.forEach((config) => {
-      const plugin = require(config.requireFile);
-      const buildInfo = plugin[config.buildUsing](config.params);
-
-      plugins.push(buildInfo.plugin);
-
-      if (typeof buildInfo.baseDir === 'function') {
-        const pluginHashForDep = hashForDep(buildInfo.baseDir());
-        cacheKeys.push(pluginHashForDep);
-      } else {
-        _logger.debug('ember-cli-htmlbars-inline-precompile is opting out of caching due to an AST plugin that does not provide a caching strategy: `' + buildInfo.name + '`.');
-        cacheKeys.push((new Date()).getTime() + '|' + Math.random());
-      }
-    });
-
-    return {
-      plugins,
-      cacheKeys
-    };
+     // Caching the pluginInfo so that call to setUp functions will be made once per worker 
+     // and not once per module tranformation
+     let key = options.parallelConfig;
+     if (!astPlugin[key]) {
```

Comment:
What is `key` here? judging from below (and the argument to `AstPlugins.setupDependentPlugins`) its an array?

Reply:
Yup. It was supposed to be hashed. Fixed it now. Please check.